handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslClientContext.java

```diff
@@ -170,6 +172,15 @@ static OpenSslSessionContext newSessionContext(ReferenceCountedOpenSslContext th
                 throw new SSLException("unable to setup trustmanager", e);
             }
             OpenSslClientSessionContext context = new OpenSslClientSessionContext(thiz, keyMaterialProvider);
+            // Enable session caching by default
+            context.setSessionCacheEnabled(true);
+            if (sessionCacheSize > 0) {
+                context.setSessionCacheSize((int) Math.min(sessionCacheSize, Integer.MAX_VALUE));
+            }
+            if (sessionTimeout > 0) {
+                context.setSessionTimeout((int) Math.min(sessionTimeout, Integer.MAX_VALUE));
```

Comment:
Should we throw if the requested `sessionCacheSize/sessionTimeout ` is higher than `Integer.MAX_VALUE`? Or at least log it? Silent override may be confusing.

Reply:
this is the same code we use for the JDK implementation. Let me think about it and make a follow up if needed to keep it consistent.